test/jdk/java/nio/channels/Channels/ReadXBytes.java

```diff
@@ -0,0 +1,311 @@
+                byte[] bytes = cis.readAllBytes();
+                assertNotNull(bytes);
+                assertEquals(bytes.length, 0);
+            }
+        );
+    }
+
+    // Verifies readAllBytes() behavior for a maximal length source
+    @Test
+    public void readAllBytesFromMaxLengthFile() throws IOException {
+        dataTest(BIG_LENGTH, (length) -> createFileWithRandomContent(length),
+            (length, cis, fis) -> {
+                byte[] cisBytes = cis.readAllBytes();
+                assertNotNull(cisBytes);
+                assertEquals(cisBytes.length, (long)length);
+                byte[] fisBytes = fis.readAllBytes();
+                assertEquals(cisBytes, fisBytes);
+            }
+        );
+    }
+
+    // Verifies readAllBytes() throws OOME if the source is too large
+    @Test(expectedExceptions = OutOfMemoryError.class)
+    public void readAllBytesFromBeyondMaxLengthFile() throws IOException {
+        dataTest(HUGE_LENGTH, (length) -> createFile(length),
+            (length, cis, fis) -> {
+                byte[] cisBytes = cis.readAllBytes();
+                assertNotNull(cisBytes);
+                assertEquals(cisBytes.length, (long)length);
+                byte[] fisBytes = fis.readAllBytes();
+                assertEquals(cisBytes, fisBytes);
+            }
+        );
+    }
+
+    // Provides an array of lengths
+    @DataProvider
+    public Object[][] lengthProvider() throws IOException {
+        return new Object[][] {
+            {1 + RAND.nextInt(1)},
+            {1 + RAND.nextInt(Byte.MAX_VALUE)},
+            {1 + RAND.nextInt(Short.MAX_VALUE)},
+            {1 + RAND.nextInt(1_000_000)},
+            {1 + RAND.nextInt(BIG_LENGTH)}
+        };
+    }
+
+    // Verifies readAllBytes() accuracy for random lengths and initial positions
+    @Test(dataProvider = "lengthProvider")
+    public void readAllBytes(int len) throws IOException {
+        dataTest(len, (length) -> createFileWithRandomContent(length),
+            (length, cis, fis) -> {
+                long position = RAND.nextInt(Math.toIntExact(length));
+                cis.skipNBytes(position);
+                byte[] cisBytes = cis.readAllBytes();
+                assertNotNull(cisBytes);
+                assertEquals(cisBytes.length, length - position);
+                fis.skipNBytes(position);
+                byte[] fisBytes = fis.readAllBytes();
+                assertEquals(cisBytes, fisBytes);
+            }
+        );
+    }
+
+    // --- readNBytes tests ---
+
+    // Verifies readNBytes() behavior for a negative length
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void readNBytesWithNegativeLength() throws IOException {
+        edgeTest(0L, (length) -> createFile(length),
+            (length, cis) -> {
+                byte[] bytes = cis.readNBytes(-1);
+                assertNotNull(bytes);
+                assertEquals(bytes.length, (long)length);
+            }
+        );
```

Comment:
It has been suggested in past PR reviews (including a few of mine) that we use assertThrows(). vs the expectedException annotation element.  Something you might want to consider using here